features/telemetry/connectors/src/main/java/org/opennms/netmgt/telemetry/connectors/ConnectorStarter.java

```diff
@@ -0,0 +1,306 @@
+        twinSubscription = twinSubscriber.subscribe(ConnectorTwinConfig.CONNECTOR_KEY, ConnectorTwinConfig.class, this::handleTwinUpdate);
+    }
+
+    private void handleTwinUpdate(ConnectorTwinConfig request) {
+        try (Logging.MDCCloseable mdc = Logging.withPrefixCloseable("telemetryd")) {
+            LOG.info("Got connectors config update - reloading");
+            synchronized (configuredLock) {
+
+                Set<String> newConfigKeys = request.getConfigurations().stream()
+                        .map(ConnectorTwinConfig.ConnectorConfig::getNodeConnectorKey)
+                        .collect(Collectors.toSet());
+                // Remove connectors that are no longer present in the new configuration
+                Set<String> currentKeys = new HashSet<>(entities.keySet());
+                for (String existingKey : currentKeys) {
+                    if (!newConfigKeys.contains(existingKey)) {
+                        delete(existingKey);
+                    }
+                }
+
+                for (ConnectorTwinConfig.ConnectorConfig config : request.getConfigurations()) {
+                    LOG.debug("Processing connector config: {}", config.getNodeConnectorKey());
+                    processConfig(config);
+                }
+            }
+        }
+    }
+
+    private void processConfig(ConnectorTwinConfig.ConnectorConfig config) {
+        String key = config.getNodeConnectorKey();
+        Entity existing = entities.get(key);
+        // start new connector
+        if (existing == null) {
+            startConnector(config);
+            return;
+        }
+
+        // if config are not changed
+        if (!hasConfigChanged(existing.config, config)) {
+            LOG.debug("No change for key: {}, skipping start", key);
+            return;
+        }
+
+        LOG.info("Configuration changed for key: {}, updating", key);
+        delete(key);
+        startConnector(config);
+    }
+
+
+    private void startConnector(ConnectorTwinConfig.ConnectorConfig config) {
+        try {
+            final Entity entity = new Entity();
+            entity.config = config;
+            final String queueName = Objects.requireNonNull(baseDef.getQueueName());
+            ensureDispatcherForQueue(queueName);
+            queueReferenceCount.incrementAndGet();
+            entity.queueName = queueName;
+            entity.connector = telemetryRegistry.getConnector(baseDef);
+
+            InetAddress ip = InetAddress.getByName(config.getIpAddress());
+            entity.connector.stream(config.getNodeId(), ip, config.getParameters());
+
+            entities.put(config.getNodeConnectorKey(), entity);
+            LOG.info("Started connector for key: {}", config.getNodeConnectorKey());
+        } catch (Exception e) {
+            LOG.error("Failed to start connector for key: {}", config.getNodeConnectorKey(), e);
+        }
+    }
+
+    private void ensureDispatcherForQueue(String queueName) {
+
```

Comment:
no need for this dispatcher logic to be so complex.

Just create dispatcher in init and use it and clean it up in stop.